### PR TITLE
Better naming and inference for PythonCallable.

### PR DESCRIPTION
**Pull request by robertwb**
This is particularly valuable for cross-language calls.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by robertwb** [comment]
R: @chamikaramj

File: sdks/python/apache_beam/utils/python_callable.py [hunk 1]
```diff
@@ -106,6 +106,19 @@ def load_from_script(source):
     exec('\n'.join(lines), exec_globals)
     return exec_globals[name]
 
+  def default_label(self):
+    src = self._source.strip()
+    last_line = src.split('\n')[-1]
+    if last_line[0] != ' ' and len(last_line) < 72:
```
Comment (chamikaramj) [inline sdks/python/apache_beam/utils/python_callable.py:112]: Where does 72 come from ?

File: sdks/python/apache_beam/transforms/util.py [hunk 2]
```diff
@@ -832,6 +833,9 @@ def from_runner_api_parameter(
 
 
 def fn_takes_side_inputs(fn):
+  if isinstance(fn, python_callable.PythonCallableWithSource):
+    # This can be removed once support for *args and **kwargs is implemented.
```
Comment (chamikaramj) [inline sdks/python/apache_beam/transforms/util.py:837]: Link to an issue ?

File: sdks/python/apache_beam/typehints/trivial_inference.py [hunk 3]
```diff
@@ -314,6 +315,9 @@ def infer_return_type(c, input_types, debug=False, depth=5):
           isinstance(input_types[1], Const)):
       from apache_beam.typehints import opcodes
       return opcodes._getattr(input_types[0], input_types[1].value)
+    elif isinstance(c, python_callable.PythonCallableWithSource):
+      # This can be removed once support for *args and **kwargs is implemented.
```
Comment (chamikaramj) [inline sdks/python/apache_beam/typehints/trivial_inference.py:319]: Ditto.

**Review by chamikaramj — COMMENTED** [review]
LGTM. Thanks.

File: sdks/python/apache_beam/transforms/util.py (hunk 2 above)
Comment (robertwb) [reply sdks/python/apache_beam/transforms/util.py]: Done.

File: sdks/python/apache_beam/typehints/trivial_inference.py (hunk 3 above)
Comment (robertwb) [reply sdks/python/apache_beam/typehints/trivial_inference.py]: Fixed.

File: sdks/python/apache_beam/utils/python_callable.py (hunk 1 above)
Comment (robertwb) [reply sdks/python/apache_beam/utils/python_callable.py:112]: An arbitrary limit for "this is getting to big to show on one line." We wouldn't want unboundedly-large stage names with huge chunks of source code in them, but if it's small like "lambda x: 2 * x" that's nice.

**Comment by robertwb** [comment]
Run Portable_Python PreCommit

**Pull request merged**